### ISSUE TITLE
Prettier status information for virtual folders created from notmuch queries

### ISSUE THREAD

**Issue by lbrayner** [neomutt/neomutt#2033]
## What would you like to be able to do?

Show prettier status information for virtual folders created from notmuch queries (\<vfolder-from-query\>, a command which I use a lot)  instead of the encoded URI.

EDIT: Also possibly a prettier and more concise folder name in the index.

## What needs to change in NeoMutt?

The %f sequence could simply show the notmuch query.

EDIT: The %b sequence in the index could show something like "<notmuch>" instead of the URI.

## Have you tried any alternative methods?

From what I could gather, pretty names are currently only possible with named virtual mailboxes, which only accept the cumbersome URI format. 

## Additional Info

![Annotation 2019-12-06 000003_blurred](https://user-images.githubusercontent.com/5733531/70395953-eac05d00-19e2-11ea-80b6-98803db68352.png)

[nm_pretty](https://github.com/lbrayner/neomutt/tree/nm_pretty) is an implementation of this idea. I patched *status.c*, *notmuch/mutt_notmuch.c*, and *notmuch/mutt_notmuch.h*. The  arguments of the notmuch URI (in `m->realpath`) are extracted and decoded with `nm_decode_uri`, a new function in *notmuch/mutt_notmuch.c*.

EDIT: I also patched *hdrline.c* to have the string "\<notmuch\>" appear for the %b sequence.

This is what it looks like upstream:

![Annotation 2019-12-06 001514_blurred](https://user-images.githubusercontent.com/5733531/70396166-25c39000-19e5-11ea-93ee-c83904a509b0.png)

**Comment by Austin-Ray** [comment]
Hi @lbrayner, thanks for opening this feature request. You've highlighted one area of inconsistency for the notmuch back-end. I appreciate it :)

I took a look at your implementation, and think I've found a simpler solution to the same goal without making the `%f` `status_format` inconsistent across back-ends. Displaying the encoded notmuch URI is intended for `%f` (same for `%b` in `index_format`)

The reason the notmuch URI appears in the `status_format` despite the `%D` is  `vfolder-from-query` does not set a name for the mailbox so it falls back to using the `%f`, which is the path (specially notmuch URI.) We can improve consistency by setting the query as the name. I've implemented it in this branch: https://github.com/neomutt/neomutt/tree/devel/issue-2033. You'll need to make sure you `status_format` contains `%D` instead of  `%f`.

---

For displaying `<notmuch>` in the `index_format` in a `%b` fashion, it's easy to do without code modifications using `folder-hook`s. Here's an implementation I came up to prefix `%b` or `<notmuch>` to my `index_format`:

```
set my_common_index_format="[%Z]  %[%Y-%m-%d]  %-20.20F  %s"
set my_non_nm_index_format="%b $my_common_index_format"
set my_nm_index_format="<notmuch> $my_common_index_format"

set index_format="$my_non_nm_index_format"

folder-hook .             "set index_format='$my_non_nm_index_format'"
folder-hook "notmuch://*" "set index_format='$my_nm_index_format'"
```
